airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -265,7 +265,7 @@ def consume_logs(*, since_time: DateTime | None = None, follow: bool = True) ->
                 time.sleep(1)
 
     def await_container_completion(self, pod: V1Pod, container_name: str) -> None:
-        while not self.container_is_running(pod=pod, container_name=container_name):
+        while self.container_is_running(pod=pod, container_name=container_name):
             time.sleep(1)
```

Comment:
While this looks like a valid fix, it is logically unrelated to the `base_container_name` change. Could you split it into another PR?

Reply:
The problem is I can't get my new unit tests for base name with get_log=False without fixing this too.

Reply:
Yeah let’s do that.

Reply:
ack. will split em tomorrow.

Reply:
@uranusjr i've split this part into: https://github.com/apache/airflow/pull/28771 ... after that merges, i'll recreate this pr.